### ISSUE TITLE
Improvement suggestions for the tutorial on flow matching

### ISSUE THREAD

**Issue by kilianFatras** [atong01/conditional-flow-matching#88]
This issue is opened for users to suggest improvements for the tutorial notebooks.

**Comment by radiradev** [comment]
Is it possible to include a minimal example of a trained model can be used to evaluate the density of a given sample?

**Comment by atong01** [comment]
@radiradev Thats a great idea. I can work on this. If you have an immediate need or are interested in making one I can send over a dirty notebook for this.

**Comment by radiradev** [comment]
@atong01 That would be very much appreciated!

**Comment by kilianFatras** [comment]
We should give credit to stochastic interpolants and rectified flows at the beginning of the notebook. Basically saying that all 3 methods are similar and concurrent ICLR2023 papers

**Comment by kilianFatras** [comment]
Hi,
Thank you! I will correct that shortly.

**Comment by radiradev** [comment]
> Is it possible to include a minimal example of how a trained model can be used to evaluate the density of a given sample?

Hi @atong01, could you please share your example notebook on how this is done?  I saw here that to evaluate the density an ode must be solved, but I am not sure how to accomplish this in pytorch:

[jax-fmx](https://github.com/albcab/fmx/blob/main/data.py)

**Comment by harveymannering** [comment]
Hello,

I cannot run the last cell in the [Flow_matching_tutorial.ipynb notebook ](https://github.com/atong01/conditional-flow-matching/blob/main/examples/2D_tutorials/Flow_matching_tutorial.ipynb) because the `sample_xt` function is not define.  Does `sample_xt` serve the same purpose as `sample_conditional_pt`?  Are the two functions interchangable?

Thank you for the tutorials!

**Comment by csufangyu** [comment]
> We should give credit to stochastic interpolants and rectified flows at the beginning of the notebook. Basically saying that all 3 methods are similar and concurrent ICLR2023 papers

Your suggestion is fantastic, I'm a beginner and it looks like there is no difference between stochastic interpolants and rectified flows, except that stochastic interpolants have an extra random term and rectified flows seem to be a deterministic sampling, is my understanding Is my understanding correct? I look forward to your reply, thanks!

**Comment by fountaindive** [comment]
Hi, this work is really cool! Was an example for how to compute the density of a generated sample ever made?

**Comment by atong01** [comment]
Yes, actually please see here https://github.com/atong01/conditional-flow-matching/blob/main/examples/2D_tutorials/Maximum_likelihood_CNF_tutorial.ipynb 

will document better and please feel free to make an issue if you have suggestions on how to make this more obvious :)

**Comment by fountaindive** [comment]
very cool! thanks!
